can not get nvme ns immediately  after create nvme ctrl

maybe need wait a sec after created nvme ctrl in nvme_test
```
======================================================================
ERROR: test_ns_info (nvme_test.NVMeTestCase.test_ns_info)
Test namespace info
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/rpmbuild/BUILD/libblockdev-3.2.0/tests/nvme_test.py", line 92, in test_ns_info
    info = BlockDev.nvme_get_namespace_info(self.nvme_ns_dev)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
gi.repository.GLib.GError: g-bd-nvme-error-quark: Failed to open device '/dev/nvme0n1': No such file or directory (1)
```

```
======================================================================
FAIL: test_connect_multiple_ns (nvme_test.NVMeFabricsTestCase.test_connect_multiple_ns)
Test connect and disconnect multiple namespaces
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/rpmbuild/BUILD/libblockdev-3.2.0/tests/nvme_test.py", line 443, in test_connect_multiple_ns
    self.assertEqual(len(namespaces), NUM_NS)
AssertionError: 1 != 3
----------------------------------------------------------------------
```
nvme-cli-2.10.2-1.el10.x86_64
nvmetcli-0.8-1.el10.noarch




```
$ git diff tests/utils.py 
diff --git a/tests/utils.py b/tests/utils.py
index a9b8e187..9fe93605 100644
--- a/tests/utils.py
+++ b/tests/utils.py
@@ -251,7 +251,7 @@ def find_nvme_ctrl_devs_for_subnqn(subnqn):
                         dev_paths += [path]
                 except:
                     pass
-
+    time.sleep(1)
     ret, out, err = run_command("nvme list --output-format=json --verbose")
     if ret != 0:
         raise RuntimeError("Error getting NVMe list: '%s %s'" % (out, err))

```
```
for i in $(seq 30);do python3 run_tests.py -s nvme_test || break;done
```
can not reproduce the error after apply the patch, maybe someone have a better solution for the error. 
if no, I will create a MR for it.

--- a/tests/nvme_test.py
+++ b/tests/nvme_test.py
@@ -416,7 +416,7 @@ class NVMeFabricsTestCase(NVMeTest):
         NUM_NS = 3
 
         # test that no device node exists for given subsystem nqn
-        ctrls = find_nvme_ctrl_devs_for_subnqn(self.SUBNQN)
+        ctrls = find_nvme_ctrl_devs_for_subnqn(self.SUBNQN, wait_for_ready=False)
         self.assertEqual(len(ctrls), 0)
 
         self._setup_target(NUM_NS)

--- a/tests/utils.py
+++ b/tests/utils.py
@@ -274,11 +274,43 @@ def clean_scsi_debug(scsi_debug_dev):
     except:
         pass
 
-def find_nvme_ctrl_devs_for_subnqn(subnqn):
+def _wait_for_nvme_controllers_ready(subnqn, timeout=3):
+    """
+    Wait for NVMe controllers with matching subsystem NQN to be in live state
+    
+    :param str subnqn: subsystem nqn to match controllers against
+    :param int timeout: timeout in seconds (default: 3)
+    """
+    start_time = time.time()
+    
+    while time.time() - start_time < timeout:
+        try:
+            for ctrl_path in glob.glob("/sys/class/nvme/nvme*/"):
+                state_file = os.path.join(ctrl_path, "state")
+                subsysnqn_file = os.path.join(ctrl_path, "subsysnqn")
+                try:
+                    state = read_file(state_file).strip()
+                    controller_subnqn = read_file(subsysnqn_file).strip()
+                    if state == "live" and controller_subnqn == subnqn:
+                        # Found a matching live controller
+                        os.system("udevadm settle")
+                        return
+                except:
+                    continue
+                
+        except:
+            pass
+        
+        time.sleep(1)
+    
+    os.system("udevadm settle")
+
+def find_nvme_ctrl_devs_for_subnqn(subnqn, wait_for_ready=True):
     """
     Find NVMe controller devices for the specified subsystem nqn
 
     :param str subnqn: subsystem nqn
+    :param bool wait_for_ready: whether to wait for controllers to be ready (default: True)
     """
 
     def _check_subsys(subsys, dev_paths):
@@ -295,6 +327,9 @@ def find_nvme_ctrl_devs_for_subnqn(subnqn):
                 except:
                     pass
 
+    # Wait for controllers to be ready if requested
+    if wait_for_ready:
+        _wait_for_nvme_controllers_ready(subnqn)
     ret, out, err = run_command("nvme list --output-format=json --verbose")
     if ret != 0:
         raise RuntimeError("Error getting NVMe list: '%s %s'" % (out, err))
@@ -317,11 +352,12 @@ def find_nvme_ctrl_devs_for_subnqn(subnqn):
     return dev_paths
 
 
-def find_nvme_ns_devs_for_subnqn(subnqn):
+def find_nvme_ns_devs_for_subnqn(subnqn, wait_for_ready=True):
     """
     Find NVMe namespace block devices for the specified subsystem nqn
 
     :param str subnqn: subsystem nqn
+    :param bool wait_for_ready: whether to wait for controllers to be ready (default: True)
     """
 
     def _check_namespaces(node, ns_dev_paths):
@@ -344,6 +380,8 @@ def find_nvme_ns_devs_for_subnqn(subnqn):
                     if 'Namespaces' in ctrl:
                         _check_namespaces(ctrl, ns_dev_paths)
 
+    if wait_for_ready:
+        _wait_for_nvme_controllers_ready(subnqn)
     ret, out, err = run_command("nvme list --output-format=json --verbose")
     if ret != 0:
         raise RuntimeError("Error getting NVMe list: '%s %s'" % (out, err))

